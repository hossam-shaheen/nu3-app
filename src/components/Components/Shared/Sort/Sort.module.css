.sort {
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.sort select {
    min-width: 150px;
    height: 35px;
    border: 1px solid #ccc;
    border-radius: 5px;
    padding: 5px;
    outline: none;
}