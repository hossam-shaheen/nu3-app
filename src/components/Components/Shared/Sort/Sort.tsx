import { FunctionComponent, useState, ChangeEvent } from 'react';
import { SortProps } from 'src/interfaces/interfaces';
import classes from './Sort.module.css';

export const Sort: FunctionComponent<{ onSort: SortProps["onSort"] }> = ({ onSort }): JSX.Element => {
    const [sort, setSort] = useState("Random");

    const onSortChange = (e: ChangeEvent<HTMLSelectElement>): void => {
        const sortKeyWord = e.target.value;
        setSort(sortKeyWord);
        onSort(sortKeyWord);
    }

    return <div className={classes.sort}>
        <select value={sort} onChange={e => onSortChange(e)} data-testid="sort-select">
            <option value={"Random"} data-testid="sort-select-option">Random</option>
            <option value={"Alphabetical"} data-testid="sort-select-option">Alphabetical</option>
            <option value={"Low"} data-testid="sort-select-option">Price: Low to High</option>
            <option value={"High"} data-testid="sort-select-option">Price: High to Low</option>
        </select>
    </div>
}
export default Sort;