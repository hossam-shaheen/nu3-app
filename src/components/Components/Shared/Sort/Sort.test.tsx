import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Sort from './Sort';

const selectedOption = (elementsIndex: number[], truthyElementIndex: number) => {
    const sortSelectOptions: any = screen.getAllByTestId('sort-select-option')
    elementsIndex.forEach((item) => {
        if (item === truthyElementIndex) {
            expect(sortSelectOptions[item].selected).toBeTruthy();
        } else {
            expect(sortSelectOptions[item].selected).toBeFalsy();
        }
    })
}

test('Sort displayed in the document: Renders', () => {
    render(<Sort onSort={jest.fn()} />);
    const sortSelect = screen.getByTestId("sort-select");
    expect(sortSelect).toBeInTheDocument();
    selectedOption([0, 1, 2, 3], 0);
});



test('Sort change to Alphabetical : Renders', () => {
    render(<Sort onSort={jest.fn()} />);
    const sortSelect = screen.getByTestId("sort-select");
    fireEvent.change(sortSelect, { target: { value: "Alphabetical" } })
    selectedOption([0, 1, 2, 3], 1);
});

test('Sort change to Random: Renders', () => {
    render(<Sort onSort={jest.fn()} />);
    const sortSelect = screen.getByTestId("sort-select");
    fireEvent.change(sortSelect, { target: { value: "Random" } })
    selectedOption([0, 1, 2, 3], 0);
});

test('Sort change to High to Low : Renders', () => {
    render(<Sort onSort={jest.fn()} />);
    const sortSelect = screen.getByTestId("sort-select");
    fireEvent.change(sortSelect, { target: { value: "Low" } })
    selectedOption([0, 1, 2, 3], 2);
});


test('Sort change to Low to High : Renders', () => {
    render(<Sort onSort={jest.fn()} />);
    const sortSelect = screen.getByTestId("sort-select");
    fireEvent.change(sortSelect, { target: { value: "High" } })
    selectedOption([0, 1, 2, 3], 3);
});

test('Sort : Snapshot', () => {
    const component = renderer.create(<Sort onSort={jest.fn()} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
