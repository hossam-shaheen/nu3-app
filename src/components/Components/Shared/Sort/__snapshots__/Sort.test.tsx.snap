// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort : Snapshot 1`] = `
<div
  className="sort"
>
  <select
    data-testid="sort-select"
    onChange={[Function]}
    value="Random"
  >
    <option
      data-testid="sort-select-option"
      value="Random"
    >
      Random
    </option>
    <option
      data-testid="sort-select-option"
      value="Alphabetical"
    >
      Alphabetical
    </option>
    <option
      data-testid="sort-select-option"
      value="Low"
    >
      Price: Low to High
    </option>
    <option
      data-testid="sort-select-option"
      value="High"
    >
      Price: High to Low
    </option>
  </select>
</div>
`;
